src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -2013,33 +1983,32 @@ bool AdaptDispatch::_ModeParamsHelper(const DispatchTypes::ModeParams param, con
         return true;
     case DispatchTypes::ModeParams::VT200_MOUSE_MODE:
         _terminalInput.SetInputMode(TerminalInput::Mode::DefaultMouseTracking, enable);
-        return !_PassThroughInputModes();
+        return true;
     case DispatchTypes::ModeParams::BUTTON_EVENT_MOUSE_MODE:
         _terminalInput.SetInputMode(TerminalInput::Mode::ButtonEventMouseTracking, enable);
-        return !_PassThroughInputModes();
+        return true;
     case DispatchTypes::ModeParams::ANY_EVENT_MOUSE_MODE:
         _terminalInput.SetInputMode(TerminalInput::Mode::AnyEventMouseTracking, enable);
-        return !_PassThroughInputModes();
+        return true;
     case DispatchTypes::ModeParams::UTF8_EXTENDED_MODE:
         _terminalInput.SetInputMode(TerminalInput::Mode::Utf8MouseEncoding, enable);
-        return !_PassThroughInputModes();
+        return true;
     case DispatchTypes::ModeParams::SGR_EXTENDED_MODE:
         _terminalInput.SetInputMode(TerminalInput::Mode::SgrMouseEncoding, enable);
-        return !_PassThroughInputModes();
+        return true;
     case DispatchTypes::ModeParams::FOCUS_EVENT_MODE:
         _terminalInput.SetInputMode(TerminalInput::Mode::FocusEvent, enable);
-        // GH#12799 - If the app requested that we disable focus events, DON'T pass
-        // that through. ConPTY would _always_ like to know about focus events.
-        return !_PassThroughInputModes() || !enable;
+        _api.GetStateMachine().InjectSequence(InjectionType::DECSET_FOCUS);
```

Comment:
A comment might be important here. Why do we need to inject one if we're currently parsing one? Is it because we want to make sure it is turned *on* even if the app just turned it off?